contracts/connection-router/tests/test.rs

```diff
@@ -107,1359 +101,1358 @@ fn route() {
     let eth = make_chain("ethereum", &mut config);
     let polygon = make_chain("polygon", &mut config);
 
-    register_chain(&mut config, &eth);
-    register_chain(&mut config, &polygon);
+    let _ = register_chain(&mut config, &eth);
+    let _ = register_chain(&mut config, &polygon);
 
     let nonce: &mut usize = &mut 0;
-    let msgs = generate_messages(&eth, &polygon, nonce, 255);
+    let messages = generate_messages(&eth, &polygon, nonce, 255);
 
     let _ = config
-        .app
-        .execute_contract(
+        .connection_router
+        .execute(
+            &mut config.app,
             eth.gateway.clone(),
-            config.contract_address.clone(),
-            &ExecuteMsg::RouteMessages(msgs.clone()),
-            &[],
+            &ExecuteMsg::RouteMessages(messages.clone()),
         )
         .unwrap();
 
-    let msgs_ret = mock::get_gateway_messages(&mut config.app, polygon.gateway, &msgs);
+    let messages_ret = mock::get_gateway_messages(&mut config.app, polygon.gateway, &messages);
 
-    assert_eq!(msgs.len(), msgs_ret.len());
-    assert_eq!(msgs, msgs_ret);
+    assert_eq!(messages.len(), messages_ret.len());
+    assert_eq!(messages, messages_ret);
 
     // try to route twice
     let res = config.app.execute_contract(
```

Comment:
why are you not using `connection_router.execute` here?

Reply:
updated